school_hostel/models/hostel.py

```diff
@@ -24,23 +26,27 @@ class HostelType(models.Model):
                              ('common', 'Common')], 'HOSTEL Type',
                             help="Type of Hostel",
                             required=True, default='common')
-    other_info = fields.Text('Other Information')
-    rector = fields.Many2one('res.partner', 'Rector')
-    room_ids = fields.One2many('hostel.room', 'name', 'Room')
+    other_info = fields.Text('Other Information',
+                             help='Enter more information')
+    rector = fields.Many2one('res.partner', 'Rector',
+                             help='Select hostel rector')
+    room_ids = fields.One2many('hostel.room', 'name', 'Room',
+                               help='Enter hostel rooms')
     student_ids = fields.One2many('hostel.student', 'hostel_info_id',
-                                  string='Students')
+                                  string='Students', help='Enter students')
 
     @api.model
     def _search(self, args, offset=0, limit=None, order=None, count=False,
                 access_rights_uid=None):
         '''Override method to get hostel of student selected'''
         if self._context.get('student_id'):
             stud_obj = self.env['student.student']
-            stud = stud_obj.browse(self._context['student_id'])
+            stud_rec = stud_obj.browse(self._context['student_id'])
             datas = []
-            if stud.gender:
-                self._cr.execute("""select id from hostel_type where type=%s or
-                type='common' """, (stud.gender,))
+            if stud_rec.gender:
+                self._cr.execute("""
+        select id from hostel_type where type=%s or type='common' """, (
+                                                    stud_rec.gender,))
```

Comment:
Here, sql injection has been used to avoid the recursive call of the search method of the same object.
But yeas, as this method has been used to get the domain, replacing this method with _name_search in following commit to do the same.

Reply:
use.. "select id .....%s" % (a,)
Thats secure.